### DISCUSSION TITLE
Welcome to Backend Internship Task's Discussions!

### DISCUSSION THREAD

**Discussion by prayashm** [DiveHQ/backend-internship-task#9] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about in QnA category.
  * Ask your doubts about the task as a new discussion.
  * ❌ Do not add a reply to this announcement.

**Comment by ankur0904** [comment]
Unable to apply from octernships page

**Reply by prayashm** [reply]
@ashwinexe can you help here?

**Reply by diivi** [reply]
Hey, can we increase the maximum number of applicants for this octernship? I believe it's 100 rn.

**Reply by prayashm** [reply]
@ashwinexe would you be able to help here?

**Reply by ashwinexe** [reply]
We are on it.

**Reply by maestroh1git** [reply]
I want to apply too. I have been working on the task, I found it through twitter. Today I checked, and a comment said to submit via github classrooms.

**Comment by subhendudash02** [comment]
Just now I have seen some of the pull requests are open in this public template. But after applying, I got an assignment from GitHub classroom in the form of a private repository. I have been using this repository for a week.

So just a minor confusion, do we need to fork this template manually, work on that and create a pull request publicly or just proceed with the private repository which automatically created by GitHub classroom? 

@prayashm kindly clarify for once.

Thank you!😊

**Reply by prayashm** [reply]
Continue using the classroom private repo and submit there when finally done.

**Reply by Abiorh001** [reply]
@prayashm hello sir, have gotten my own private repo and created a separate branch. how do I submit the assignment so it can be visible to the organization. have already create a pull request from my private repo that was generated for me by your organization.

**Reply by prayashm** [reply]
You need to create the repo from Github Classroom.
At the moment the capacity is being increased to submit via Classroom.
Until then please wait.

**Reply by Abiorh001** [reply]
yes the repo was created for me through the GitHub classroom bot and I was given a private repo and I already add another branch and make a pull request. I only want to know if I have submitted the assignment like that or I still need another steps to take.

**Reply by prayashm** [reply]
I see your private repo.
Your code should be merged into `main`. 
You can close the PR#2 called `final submission`.

There's a Feedback PR#1 (already open) where we'll review.

**Reply by subhendudash02** [reply]
Okay, so we need to push the changes only in `feedback` branch and the PR#1 will be automatically updated right?
Even I made the new branch and I was going to make a new PR as per the instructions given in the README referencing github flow.

**Reply by Abiorh001** [reply]
I already merge the final submission to the main repo and closed the pull request. I want to know if I have successfully submitted the assignment and I'm good to go or I need to do another thing. thank you sir for your responds

**Reply by ashwinexe** [reply]
@subhendudash02 

Hey there, PRs to template repository are not valid submissions to Octernships. Only your private repository submissions are considered for evaluation.

**Reply by ashwinexe** [reply]
> so we need to push the changes only in feedback branch and the PR#1 will be automatically updated right?

@subhendudash02 if you look closely, the feedback PR points from `main` to `feedback` and is only for maintainers to provide you **feedback**. 

Your submission is considered when you commit to `main` so please create your own branch (with PR from `branch` -> `main`) for submission (aka The GitHub flow)

**Comment by Abiorh001** [comment]
please i cannot find any link to the github classroom so as to submit my assignment

**Reply by ashwinexe** [reply]
> Can you try this: [redacted]

@prayashm please avoid sharing classroom invite link publicly. We're working to increase capacity of your project.

Highly advised to remove this invite

**Reply by prayashm** [reply]
Sure.

**Reply by narayan954** [reply]
please remove the revision from edit history as well. I applied through the edit history just now. I was trying to apply through octernship portal but it's crashing there, maybe there's a bug in that invite link present in the portal. Whenever I click on apply it redirects me back to the same page.

**Reply by prayashm** [reply]
Yup, we deactivated the link from the edit history.
You should apply from the Octernship portal -> otherwise it might not be considered by Github.
Capacity to accept was increased today so you would be able to apply now.

**Reply by narayan954** [reply]
thanks for confirming :)

**Comment by Abiorh001** [comment]
@subhendudash02 how do i get access to the private repo so i can submit it via github classroom

**Reply by ashwinexe** [reply]
You have to apply via https://education.github.com/students/octernships

**Comment by subhendudash02** [comment]
> > so we need to push the changes only in feedback branch and the PR#1 will be automatically updated right?
> 
> @subhendudash02 if you look closely, the feedback PR points from `main` to `feedback` and is only for maintainers to provide you **feedback**.
> 
> Your submission is considered when you commit to `main` so please create your own branch (with PR from `branch` -> `main`) for submission (aka The GitHub flow)

Ah, just now I observed where that PR is pointing to, I thought other way round (feedback to main). Got it, thanks!

**Comment by harshk30** [comment]


**Reply by prayashm** [reply]
A1: Depends of quality of submissions we get.
A2: You need to apply via [education.github.com/students/octernships](https://education.github.com/students/octernships)

**Comment by Gibril1** [comment]
I submitted a PR on this repo but I was told to apply on github classrooms. I have done that and I am waiting for approval, does it take long to get approval. I am done with the project and I want to submit

**Comment by suyogkokaje** [comment]
Till now I have directly commited my changes to the main branch of the private repository generated by the github classroom for me. So what should I do now? Should I create a new branch and then raise a PR from it for submission? Please help. Thank you.

**Reply by ashwinexe** [reply]
> Till now I have directly commited my changes to the main branch of the private repository generated by the github classroom for me. So what should I do now? Should I create a new branch and then raise a PR from it for submission? Please help. Thank you.

That's okay @suyogkokaje. The purpose of PR is for maintainers to have clear visibility on changes you've done.

**Comment by AryanMadhavVerma** [comment]
I had applied for GitHub education a day back, to get access to the classroom where we have to submit the task. Verification is still pending and hence access is restricted. I'm updating my progress on a forked repository with periodic commits. Is that alright? Or will DiveHQ only take submissions on the classroom portal? 
@ashwinexe @prayashm

**Reply by prayashm** [reply]
We'll take submissions only from the classroom.

**Comment by Shurtu-gal** [comment]
@ashwinexe, Can you give me some feedback if I am going the correct way till now?

**Comment by nikhil25803** [comment]
Can anyone from the admin confirm my submission 👀
Just needed a confirmation.

**Comment by suyogkokaje** [comment]
Is there any specific procedure I have to do in order to submit ? I have pushed my code on my classroom repository.

**Comment by swapnalshahil** [comment]
Hello maintainers! I feel like there is a bug on octernship page as I applied for 3 projects and I got private repos from GitHub Classroom and I can see only one project on the Your projects section. Anyways I worked on the DiveHQ backend-internship-task and pushed code by raising a PR through a branch which points to main. I still need to add test cases, then it is good to go. Just want to know if you guys can see my work there? Also submission is accepted through PR or should I have to merge that PR?

**Reply by 0xkarambit** [reply]
I thought i was the only one with that bug, btw would you happen to know how long we have left to complete the submission ? i know the deadline is 19th but i didnt find a time anywhere.

**Reply by swapnalshahil** [reply]
> I thought i was the only one with that bug, btw would you happen to know how long we have left to complete the submission ? i know the deadline is 19th but i didnt find a time anywhere.

It's Jun 19, 2023, 18:00 IST

**Reply by 0xkarambit** [reply]
Thank You ! @swapnalshahil

**Reply by swapnalshahil** [reply]
@prayashm can you see my submission? I raised a PR on my private repo created by Github classroom

**Comment by Shurtu-gal** [comment]
Hey @maestroh1git @swapnalshahil is there any specific way to get feedback

**Comment by shuklaritvik06** [comment]
Hey @prayashm I have completed my submission and asked for your review

**Comment by idyweb** [comment]
hi i have applied and my application has been approved, what next? Because i cant see what else to do

**Reply by shuklaritvik06** [reply]
Hey @idyweb  👋🏼  After successfully applying you must have got the email from github classroom inviting you to the private repo. You have to use that repo for submission purposes.

### Steps to Follow

- [x] Create a new branch `dev` from the main branch
- [x] Commit your changes to the `dev` branch
- [x] Now raise a PR from `dev` to `main`
- [x] Merge that PR
- [x] Your changes would be visible in Feedback#1 PR and there folks will review it

Deadline is Today 18:00 PM IST

**Reply by ashwinexe** [reply]
@idyweb I can confirm you are not assigned this Octernship, can you double check if you applied correctly? Screenshot can help me understand this better.

**Comment by idyweb** [comment]
Hi i havent gotten an email from github classroom yet

**Comment by shuklaritvik06** [comment]
> Hi i havent gotten an email from github classroom yet

Check the notifications once

**Reply by idyweb** [reply]
no notification either bro.. i am so tensed right now.

**Reply by shuklaritvik06** [reply]
Hey @prayashm  @ashwinexe  can you please help him

**Reply by ashwinexe** [reply]
Got my 👀. 
Pro Tip: Highly recommended to create a new discussion in here to separate conversations.

**Comment by Shurtu-gal** [comment]
What do we have to do now?
Do we need to ping someone in the feedback PR?

**Comment by shuklaritvik06** [comment]
> What do we have to do now? Do we need to ping someone in the feedback PR?

You can ask for review from the folks of DiveHQ

**Reply by Shurtu-gal** [reply]
Can you elaborate on the procedure?

**Comment by swapnalshahil** [comment]
Did anyone get feedback? 👀

**Reply by shuklaritvik06** [reply]
Nope Not Yet

**Reply by 0xkarambit** [reply]
Nope

**Reply by Shurtu-gal** [reply]
Nope
